common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/DbSchemaCreator.java

```diff
@@ -142,18 +153,20 @@ private ScriptRunner getScriptRunnerInstance(Connection connection) {
     return runner;
   }
 
-  private boolean isSchemaPreexisting(Connection connection) {
+  private boolean isSchemaPreexisting(Connection connection, String dbProductId) {
     ScriptRunner runner = getScriptRunnerInstance(connection);
     StringWriter errorWriter = new StringWriter();
     runner.setErrorLogWriter(new PrintWriter(errorWriter));
-    try {
-      String scriptPath =
-          selectDbSchemaDetectionScript(connection.getMetaData().getDatabaseProductName());
-      InputStream resourceAsStream = DbSchemaCreator.class.getResourceAsStream(scriptPath);
-      BufferedReader reader =
-          new BufferedReader(new InputStreamReader(resourceAsStream, StandardCharsets.UTF_8));
+
+    String scriptPath = selectDbSchemaDetectionScript(dbProductId);
+
+    try (InputStream resourceAsStream = DbSchemaCreator.class.getResourceAsStream(scriptPath);
+        InputStreamReader inputReader =
+            new InputStreamReader(resourceAsStream, StandardCharsets.UTF_8);
+        BufferedReader reader = new BufferedReader(inputReader)) {
       runner.runScript(getSqlSchemaNameParsed(reader));
-    } catch (RuntimeSqlException | SQLException e) {
+
+    } catch (RuntimeSqlException | IOException e) {
```

Comment:
I think this can be formatted a little better. Can you have another look please?
e.g. The InputStreamReader can probably be reduced to a single line when `resourceAsStream` gets a shorter name.